File: Swift-C++-Testsuite/test/error-handling/Swift/ErrorToOptional.swift
@main
struct MyMain {

    static func main() {
        func someThrowingFunction() throws -> Int {
            // ...
            return 42
        }

        let x = try? someThrowingFunction()

        let y: Int?
        do {
            y = try someThrowingFunction()
        } catch {
            y = nil
        }

    }

}

File: Swift-C++-Testsuite/test/error-handling/C++/ErrorToOptional.cpp
#include <iostream> 
using namespace std;

        static int someThrowingFunction() noexcept(false) {
            // ...
            return 42;
        }

int main() {

        int y;
        try {
            y = someThrowingFunction();
        } catch(exception &ex) {
            y = -1;
        }
    }

